core/templates/dev/head/pages/exploration_editor/editor_tab/StateHints.js

```diff
@@ -149,8 +151,19 @@ oppia.controller('StateHints', [
           }
         ]
       }).result.then(function() {
-        stateHintsService.displayed.splice(index, 1);
-        stateHintsService.saveDisplayedValue();
+        if (stateSolutionService.displayed) {
```

Comment:
I agree -- just do the change directly, and then, later, do a separate if/else check for the modal.